### ISSUE TITLE
Not all Heroes are listed

### ISSUE THREAD

**Issue by mmichau** [suurjaak/h3sed#1]
Hi, Great tool but I find an issue that not all heroes are listed. For example, from all heroes that I play, I see only 2/5 :/

**Comment by suurjaak** [comment]
I see I have forgotten to update this project for a while; this bug could already be fixed, along with a number of other major improvements. I have released version 1.6 now, you can try that and see how it fares. 

Thank you for prodding me.

**Comment by mmichau** [comment]
Great! Thanks for releasing a new version. Unfortunately the issue that I reported is still there :/

**Comment by suurjaak** [comment]
Can you share the savefile?  What game version is this from?

**Comment by mmichau** [comment]
It's a random HOTA map. Save is here: https://we.tl/t-6XLcedgX8k 

Example: Heroes with name **Ingham** or **Astral** are not listed

**Comment by suurjaak** [comment]
Found the problem, spell scrolls in inventory are not interpreted correctly. 

Astral has this in inventory, right?

```yaml
inventory:
- Dragonbone Greaves
- Spell Scroll: View Air
- Spellbinder's Hat
- Spell Scroll: Dimension Door
- Spell Scroll: Summon Water Elemental
- Endless Sack of Gold
- Celestial Necklace of Bliss
- Hellstorm Helmet
- Plate of Dying Light
```

And Ingham has this?
```yaml
inventory:
- Speculum
- Spell Scroll: Earthquake
- Crown of the Supreme Magi
- Spell Scroll: Implosion
- Spell Scroll: Curse
- Spell Scroll: View Earth
- Vampire's Cowl
- Spell Scroll: Stone Skin
- Spell Scroll: Protection from Water
- Spell Scroll: Curse
- Spell Scroll: Hypnotize
- Spell Scroll: Frost Ring
```

**Comment by mmichau** [comment]
Yes, both have a lot of scrolls.

**Comment by suurjaak** [comment]
I can send you a pre-release version so you could test whether it all works properly now, if you're interested. 

Which program release were you using, the Windows binaries?

**Comment by mmichau** [comment]
Yes, please. I will be able to test this tonight/tomorrow.

Windows x64 binary will be good.

**Comment by suurjaak** [comment]
The stand-alone executable: https://erki.lap.ee/stuff/h3sed_1.7.dev2_x64.exe

**Comment by suurjaak** [comment]
@mmichau any luck?

**Comment by mmichau** [comment]
I still don't see all heroes. I even clean one hero and from him, I move all Spell scroll items to another hero (a new one) and this one was visible. 
Conclusion: heroes with scrolls now are visible but still there is another issue that blocks listing all heroes.

In the meantime, I also noticed another issue. Modifying the movement points works only for the first turn. After the first turn, everything goes back to default values. I tried several times(with different values) but without success.

Do you want to get a new save file?

**Comment by suurjaak** [comment]
Can you tell the names of some of the heroes still missing? 

> Modifying the movement points works only for the first turn.

Confirmed, looks like the game overwrites these values upon ending the turn. Bummer. I guess I'll have to make the "movement points in total" a read-only setting in the editor, or remove it altogether, as modifying it is pointless.

**Comment by suurjaak** [comment]
@mmichau Hello?

**Comment by mmichau** [comment]
@suurjaak Sorry, my PC died and had no access to the game and github for some time. I will check the names and post them this evening.

**Comment by suurjaak** [comment]
Ouch, sorry to hear that.

**Comment by suurjaak** [comment]
@mmichau Ping?

**Comment by mmichau** [comment]
Apologize for such a late reply.
Still missing Heroes: Aine, Adela, Astral

**Comment by suurjaak** [comment]
Interesting.. This is with the "1.7.dev2" version I linked on May 17th? And from the same savegame? 

Because this is what it shows for me now for that savegame: [Heroes from test.html.zip](https://github.com/suurjaak/h3sed/files/11713131/Heroes.from.test.html.zip). 179 heroes, and Ingham, Astral, Aine and Adela are all there.

**Comment by mmichau** [comment]
Hmm, strange...
I use h3sed_1.7.dev2_x64.exe

Can you try this save file:
[AUTOZAPIS_TEST.zip](https://github.com/suurjaak/h3sed/files/11713308/AUTOZAPIS_TEST.zip)

**Comment by suurjaak** [comment]
With that savegame I get the same result, 3 heroes missing. Seems HotA has started using some byte positions for unknown purposes and that is why these heroes do not get recognized. 

Is this what Astral looks like in-game currently?
```yaml
Astral:
  stats:
    attack:          32
    defense:         25
    power:           29
    knowledge:       31
    exp:             237206
    level:           25
    movement_total:  2240
    movement_left:   2240
    mana:            465
    spellbook:       true
    ballista:
    ammo:            false
    tent:            true
  artifacts:
    helm:            Hellstorm Helmet
    neck:
    armor:
    weapon:          Red Dragon Flame Tongue
    shield:          Buckler of the Gnoll King
    lefthand:        Equestrian's Gloves
    righthand:       Equestrian's Gloves
    cloak:           Angel Wings
    feet:            Sandals of the Saint
    side1:
    side2:
    side3:
    side4:
    side5:
```

**Comment by mmichau** [comment]
Yes, That is Astral, He is one of the heroes that was not visible from the beginning.

**Comment by suurjaak** [comment]
How does it look with this version: https://erki.lap.ee/stuff/h3sed_1.7.dev6_x64.exe ?

**Comment by suurjaak** [comment]
@mmichau Ping?

**Comment by mmichau** [comment]
Hello, I'm on holidays. Will be back after 3-07 with access to the
computer.

I will check everything and let a comment.

Greetings,
Michał

pon., 19 cze 2023, 17:23 użytkownik Erki Suurjaak ***@***.***>
napisał:

**Comment by suurjaak** [comment]
@mmichau Back from holidays?

**Comment by suurjaak** [comment]
So, at least the original issue is resolved, all heroes are now listed in the editor?

**Comment by suurjaak** [comment]
@mmichau Ping? I would really like to make a new public release for this already.

**Comment by mmichau** [comment]
@suurjaak All heroes are now listed. The original issue is resolved. Thank you !

**Comment by suurjaak** [comment]
Excellent, glad to hear it. I have now released v1.7 with these fixes.
